mesonbuild/compilers.py

```diff
@@ -1953,6 +1953,9 @@ def unix_link_flags_to_native(self, args):
                     continue
                 else:
                     i = name + '.lib'
+            # -pthread in link flags is only used on Linux
```

Comment:
`-lpthread` also needs to be filtered out. Also the current code does fewer comparisons in the usual case.